botnim/collect_sources.py

```diff
@@ -1,53 +1,117 @@
 import io
-import pathlib
+from pathlib import Path
 import dataflows as DF
-
+from datetime import datetime, timezone
+import json
+from content_extraction import extract_structured_content
 
 def collect_sources_files(config_dir, context_name, source):
     files = list(config_dir.glob(source))
-    file_streams = [(f.name, f.open('rb'), 'text/markdown') for f in files]
+    file_streams = []
+    for f in files:
+        with open(f, 'rb') as ff:
```

Comment:
1. Why read the files and not keep them as file streams?
2. This is exactly the place to check for an accompanying `.metadata.json` file, read and parse it and add the metadata to the filestreams.